runtime/doc/news.txt

```diff
@@ -80,6 +80,14 @@ The following changes may require adaptations in user config or plugins.
   the option is set (e.g. using |:set| or |nvim_set_option_value()|) without a
   scope, which means they now behave the same way as string options.
 
+• Signs placed through the legacy |sign-commands| are now stored and displayed
+  as |extmarks| internally. Along with the following changes:
+  • Lines deleted and restored with |undo| have their signs removed and restored.
```

Comment:
I think this can be considered a breaking change and will require me to make changes to several of my plugins.

Reply:
yea, if we add the "ignore undo" mode (I'll put up a PR), it would make sense to use that for legacy signs.

Reply:
Ok so what do we want to happen in this PR upon line deletion?

1. Both legacy and extmark signs will be deleted from the buffer (regardless of 'signcolumn', "breaking" for extmark signs).
2. We make a superficial distinction between extmark and legacy signs and have extmark signs adjusted and legacy signs deleted, just to stay backwards compatible.
3. Both legacy and extmark signs will be adjusted instead of deleted ("breaking" for legacy signs).

Reply:
The distinction will rather be that extmark signs _explicitly_ can choose either mode, just like any other extmark.

"legacy" behavior becomes a strict subset of the configurable extmark behavior in either case.

Reply:
Yeah fair, I guess we're set then once I rebase this on top of #25882. Legacy signs will have both `invalidate` and `no_undo_restore` set and will be deleted, only breaking the current `'signcolumn' == yes/auto:1` behavior.